### DISCUSSION TITLE
TreeSelect/Tree component large data set performance

### DISCUSSION THREAD

**Discussion by rdyava** [primefaces/community#1299] [PrimeReact] [answered]
Hello, I am running into issues with a large amount of nodes in my dataset (30,000+) and would like to know if there is an amount of nodes that the component can support with sub 1 second filter results? Alternatively, if using lazy loading with the entire dataset, would the filter work without issue, or would there be logic that needs to be implemented for the filter to support lazy loading and search through all nodes?

**Comment by melloware** [comment, marked as answer]
Yes you want to use the `lazy` example and only load data as necessary.  It is for this exact scenario: https://primereact.org/tree/#lazy
